tests/test_streaming_manager.py

```diff
@@ -0,0 +1,193 @@
+import time
+from datetime import datetime
+from unittest.mock import MagicMock, Mock
+
+import pytest
+import requests
+
+from flagsmith import Flagsmith
+from flagsmith.exceptions import FlagsmithAPIError
+from flagsmith.streaming_manager import EventStreamManager
+
+
+def test_stream_not_used_by_default(requests_session_response_ok, server_api_key):
+    flagsmith = Flagsmith(
+        environment_key=server_api_key,
+        enable_local_evaluation=True,
+    )
+
+    assert hasattr(flagsmith, "event_stream_thread") is False
+
+
+def test_stream_used_when_enable_realtime_updates_is_true(
+    requests_session_response_ok, server_api_key
+):
+    flagsmith = Flagsmith(
+        environment_key=server_api_key,
+        enable_local_evaluation=True,
+        enable_realtime_updates=True,
+    )
+
+    assert hasattr(flagsmith, "event_stream_thread") is True
+
+
+def test_error_raised_when_realtime_updates_is_true_and_local_evaluation_false(
+    requests_session_response_ok, server_api_key
+):
+    with pytest.raises(ValueError):
+        Flagsmith(
+            environment_key=server_api_key,
+            enable_local_evaluation=False,
+            enable_realtime_updates=True,
+        )
+
+
+def test_stream_manager_handles_timeout(mocked_responses):
+    stream_url = (
+        "https://realtime.flagsmith.com/sse/environments/B62qaMZNwfiqT76p38ggrQ/stream"
+    )
+
+    mocked_responses.get(stream_url, body=requests.exceptions.ReadTimeout())
+
+    streaming_manager = EventStreamManager(
+        stream_url=stream_url,
+        on_event=MagicMock(),
+        daemon=True,
+    )
+
+    streaming_manager.start()
+
+    time.sleep(0.01)
+
+    assert streaming_manager.is_alive()
+
+    streaming_manager.stop()
+
+
+def test_stream_manager_handles_request_exception(mocked_responses, caplog):
+    stream_url = (
+        "https://realtime.flagsmith.com/sse/environments/B62qaMZNwfiqT76p38ggrQ/stream"
+    )
+
+    mocked_responses.get(stream_url, body=requests.RequestException())
+    mocked_responses.get(stream_url, body=FlagsmithAPIError())
```

Comment:
Putting exceptions in the body here seems odd, is this some feature of the updated version of `responses` ?

Reply:
It does look a bit odd doesn't it, but it is a feature https://github.com/getsentry/responses/blob/master/README.rst#exception-as-response-body

Reply:
How strange (but quite useful!)